$NetBSD: patch-ad,v 1.2 1999/03/07 15:44:06 hubertf Exp $

Submitted by Thilo Manske <Thilo.Manske@HEH.Uni-Oldenburg.DE> after
some discussion on PR 7093.


--- screens.c.BAK	Sun Mar  7 16:32:59 1999
+++ screens.c	Sun Mar  7 16:36:03 1999
@@ -33,6 +33,7 @@
     endwin();
     clear();
     refresh();
+    reset_shell_mode();
     if (sig != 0)
        exit(1);
 }
@@ -64,17 +65,17 @@
    mvaddstr(Y+1,X+8,"Left"); 
    mvaddstr(Y+2,X+8,"Right"); 
    attrset(COLOR_PAIR(COLOR_SLIDER) | A_ALTCHARSET);
-   mvaddstr(Y+1,X+14,"อออออออออออออออออออออ"); 
-   mvaddstr(Y+2,X+14,"อออออออออออออออออออออ"); 
+   mvaddstr(Y+1,X+14,"---------------------");
+   mvaddstr(Y+2,X+14,"---------------------");
    attrset(COLOR_PAIR(COLOR_VOL1) | A_ALTCHARSET);
-   mvaddstr(Y,X+14,"๙ ๙ ๙ ๙");
-   mvaddstr(Y+3,X+14,"๙ ๙ ๙ ๙");
+   mvaddstr(Y,X+14,". . . .");
+   mvaddstr(Y+3,X+14,". . . .");
    attrset(COLOR_PAIR(COLOR_VOL2) | A_ALTCHARSET);
-   mvaddstr(Y,X+22,"๙ ๙ ๙ ๙");
-   mvaddstr(Y+3,X+22,"๙ ๙ ๙ ๙");
+   mvaddstr(Y,X+22,". : . .");
+   mvaddstr(Y+3,X+22,". : . .");
    attrset(COLOR_PAIR(COLOR_VOL3) | A_ALTCHARSET);
-   mvaddstr(Y,X+30,"๙ ๙ ๙");
-   mvaddstr(Y+3,X+30,"๙ ๙ ๙");
+   mvaddstr(Y,X+30,". . .");
+   mvaddstr(Y+3,X+30,". . .");
 }
 
 /*
@@ -87,26 +88,27 @@
 {
    if ((level == 1 ) || ( level == 2 )) { 
       attrset(A_NORMAL  | A_ALTCHARSET); 
-      mvaddch(Y,X++,205); 
+      mvaddch(Y,X++,'-');
    } 
    if ( level != 5) { 
-      #define ACT_HANDLE COLOR_PAIR(COLOR_HANDLE) | A_BOLD | A_ALTCHARSET
-      #define NACT_HANDLE COLOR_PAIR(COLOR_HANDLE) | A_DIM | A_ALTCHARSET    
+      #define ACT_HANDLE COLOR_PAIR(COLOR_HANDLE) | A_BOLD
+      #define NACT_HANDLE COLOR_PAIR(COLOR_HANDLE) | A_DIM
       attrset(( active == TRUE  ? ACT_HANDLE : NACT_HANDLE)); 
-      mvaddch(Y,X++,178); 
+      mvaddch(Y,X++,(active ? 'O' : 'o'));
    } 
    if ((level == 3) || (level == 2)) { 
       attrset(A_NORMAL | A_ALTCHARSET); 
-      mvaddch(Y,X++,205); 
+      mvaddch(Y,X++,'-');
    } 
    if ( level  == 5 ) { 
       attrset(A_NORMAL | A_ALTCHARSET); 
-      mvaddch(Y,X++,205); 
+      mvaddch(Y,X++,'-');
    }    
 }
 
 void init_screen(void)
 {
+     def_shell_mode();
      stdscr = initscr();
      if ( (COLS < 80) || (LINES < 25) ) {
         finish(0);
@@ -181,9 +183,9 @@
    char ch; 
    touchwin(stdscr); 
    help_win = newwin(15,50,(LINES - 15) / 2,(COLS - 50) / 2);
-   wattrset(help_win, COLOR_PAIR( COLOR_HELP) | A_BOLD | A_ALTCHARSET); 
+   wattrset(help_win, COLOR_PAIR( COLOR_HELP) | A_BOLD );
    leaveok(help_win,TRUE); 
-   wborder(help_win,179,179,196,196,218, 191,192,217); 
+   wborder(help_win,'|','|','-','-','+','+','+','+');
    wattrset(help_win, A_NORMAL | A_BOLD);
    mvwaddstr(help_win,2,2," Left/Right : Lower/Higher "); 
    mvwaddstr(help_win,3,2," Up/Down    : Previous/Next "); 
